pkg/controller/elasticsearch/nodespec/resources.go

```diff
@@ -54,7 +55,8 @@ func BuildExpectedResources(
 		if nodeSpec.Config != nil {
 			userCfg = *nodeSpec.Config
 		}
-		cfg, err := settings.NewMergedESConfig(es.Name, ver, ipFamily, es.Spec.HTTP, userCfg)
+		isMonitoringMetrics := stackmon.IsMonitoringMetricsDefined(es)
+		cfg, err := settings.NewMergedESConfig(es.Name, ver, ipFamily, es.Spec.HTTP, userCfg, isMonitoringMetrics)
```

Comment:
Feels a bit odd how the "knowledge" about stack monitoring is spread here between this place and the `NewMergedESConfig` I wonder if would be cleaner to just pass in an additional (sometimes empty) config object?

```go
cfg, err := settings.NewMergedESConfig(es.Name, ver, ipFamily, es.Spec.HTTP, userCfg, stackmon.MonitoringConfig())
```